zebra-chain/src/serialization/constraint.rs

```diff
@@ -165,7 +171,28 @@ impl<T> From<AtLeastOne<T>> for Vec<T> {
     }
 }
 
+// `IntoIterator<Item = T>`, but not `Item = &mut T`, because that could break the constraint
```

Comment:
I think it's not possible to remove an item using `Iterator<Item = &mut T>`, so I think it wouldn't break the constraint :thinking:

Reply:
Fixed in 3491e01cc